lib/routes/51cto/rss.ts

```diff
@@ -0,0 +1,48 @@
+import { Route } from '@/types';
+import { parseDate } from '@/utils/parse-date';
+import got from '@/utils/got';
+import { load } from 'cheerio';
+
+export const route: Route = {
+    path: '/',
+    categories: ['programming'],
+    example: '/51cto',
+    radar: [
+        {
+            source: ['51cto.com/'],
+        },
+    ],
+    name: '推荐',
+    maintainers: ['cnkmmk'],
+    handler,
+    url: '51cto.com/',
+};
+
+async function handler() {
+    const url = 'https://51cto.com';
+    const response = await got({ method: 'get', url });
+    const $ = load(response.data);
+
+    const list = $('div.article-irl-c.split-left-l')
+        .map((i, e) => {
+            const element = $(e);
+            const title = element.find('a.usehover.article-irl-ct_title').text();
+            const link = element.find('a.usehover.article-irl-ct_title').attr('href');
+            const description = element.find('a.split-top-m.usehover.pc-three-line.article-abstract').text();
+            const dateraw = element.find('p.article-irl-cb_time').text();
+
+            return {
+                title,
+                description,
+                link,
+                pubDate: parseDate(dateraw, 'YYYY-MM-DD HH:mm:ss'),
+            };
+        })
+        .get();
```

Comment:
Please use the API from `/index/index/recommend` instead.

Reply:
试了一下，API好像是无效。

Reply:
The API does work.

<details><summary>Screenshot</summary>

![image](https://github.com/DIYgod/RSSHub/assets/11386903/7456d7b4-cfc7-4aa1-aac8-659abc121483)
</details>

Reply:
已经修正了，现在的路径是 /51cto/index/recommend

Reply:
The code is still using HTML elements instead of the API endpoint from `/index/index/recommend`
https://github.com/DIYgod/RSSHub/blob/17d808cd007133b0399af63ff3a0979f15b0f785/lib/routes/51cto/recommend.ts#L24-L32

Reply:
是必须要使用 https://www.51cto.com/index/index/recommend 这个API，而不能直接使用HTML代码吗？
这个API，我无法访问的。

Reply:
Of course you can't access `www.51cto.com/index/index/recommend` since the API host is `api-media.51cto.com`. If an API is available, you need to use that instead of going through HTML.

Reply:
已经调整，使用API生成RSS。